commons/zenoh-shm/src/posix_shm/cleanup.rs

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+pub(crate) use platform::cleanup_orphaned_segments;
+
+#[cfg(not(unix))]
+mod platform {
+    pub(crate) fn cleanup_orphaned_segments() {}
```

Comment:
Maybe we can add a `tracing::warn` in case `cleanup_orphaned_segments` does nothing?

Reply:
Well, no, because what I want is to abstract our users from the platform-specific details of our implementation (that even might change in the future!)